### PR TITLE
Added CASTEP example

### PR DESCRIPTION
**Pull request by mijp1**
CASTEP v22 onwards has nature support for sockets and hence can be used with i-pi.
This is a trimmed down example of a 4-bead PIMD run using i-pi and CASTEP, with a simple script to automate launch & execution.

### CODE REVIEW

**Comment by mijp1** [comment]
Note: this PR does not change any of the i-pi source - it is just an addition to the "examples" showing how CASTEP v22 onwards now works natively with i-pi

File: examples/castep/run_me [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+#all present so now all OK to launch i-pi
+set nbeads = `grep nbead ${seed}.xml | awk -F\' '{print $2}'`
+if ( "$nbeads" == "" ) then
+   set nbeads = `grep nbead ${seed}.xml | awk -F\" '{print $2}'`
+endif
+
+echo Starting i-pi with $nbeads beads for system $seed
+i-pi ${seed}.xml > log &
+
```
Comment (litman90) [inline examples/castep/run_me:56]: I would add a  'sleep 5' here since in some machines i-pi takes a  few seconds to initialize.

**Review by litman90 — CHANGES_REQUESTED** [review]

**Comment by litman90** [comment]
Thanks for the contribution!
I activated the automatic test which should pass without problems. Please add the line I suggested. Afterwards, we are ready to merge

**Comment by ceriottm** [comment]
thanks @mijp1 for the contribution, and thanks Yair for vetting it. I'll
also add a reference to CASTEP among the list of codes that support the
i-PI interface.
Best
Michele

File: examples/castep/run_me (hunk 1 above)
Comment (mijp1) [reply examples/castep/run_me:56]: OK - now done.
    Matt

**Review by litman90 — APPROVED** [review]

**Pull request merged**